herbert_core/classes/data_loaders/@loader_nxspe/loader_nxspe.m

```diff
@@ -245,6 +265,90 @@
                 obj = set_data_file_name@a_loader(obj,filename);
             end
         end
+        function instrument = read_instrument_info_(obj)
+            if ~isempty(obj.file_name)
+                filename = obj.file_name;
+            else
+                filename = obj.par_file_name;
+            end
+            if ~isempty(obj.root_nexus_dir)
+                root_dir = obj.root_nexus_dir;
+            else
+                root_dir = find_root_nexus_dir(filename);
+            end
+            h5inst = h5info(filename, [root_dir '/instrument']);
+            dataset = read_nexus_groups_recursive(h5inst);
+            % Instrument we support must have 'moderator' and 'source' components
+            if ~any(isfield(dataset, {'moderator', 'source'}))
+                error('HERBERT:loader_nxspe:invalid_instrument', ...
+                      'nxspe file has instrument data incompatible with Horace');
+            end
+            source = IX_source(dataset.source.Name.value, '', double(dataset.source.frequency.value));
+            moderator = obj.read_inst_moderator_(dataset);
+            % NXSPE files *must* have a "fermi" component, so we distinguish
+            % instrument types based on presence of 'shaping_chopper' and 'mono_chopper'
+            % as we only have two types of instruments supported at present
+            errmsg = ['nxspe file has incomplete instrument data. Please manually set ' ...
+                       'instrument data if you want to perform resolution convolution.'];
+            if all(isfield(dataset, {'shaping_chopper', 'mono_chopper'}))
+                % The struct must have also have 'horiz_div', and 'vert_div' fields
+                if ~any(isfield(dataset, {'horiz_div', 'vert_div'}))
+                    error('HERBERT:loader_nxspe:missing_instrument_fields', errmsg);
+                end
+                instrument = obj.read_disk_inst_(dataset, source, moderator);
+            else
+                % The struct must have also an 'aperture' field
+                if ~isfield(dataset, 'aperture')
+                    error('HERBERT:loader_nxspe:missing_instrument_fields', errmsg);
+                end
```

Comment:
Yes, I think the previous logic was a bit convoluted (I have not idea what I was thinking then). The requirements for an `nxspe` file to have valid instrument information as far as Horace is concerned is that:

* All files must have a `moderator` and `source`.
* Disk chopper instruments must have `shaping_chopper`, `mono_chopper`, `horiz_div` and `vert_div` (this will create an `IX_inst_DGdisk`
* Fermi chopper instruments must have `aperture` (this will create an `IX_inst_DGfermi`.

Previously I was looking at the presence of the two choppers to check if the user wanted to define a disk or Fermi chopper instrument and then erroring if they did not specify the divergences but really I should have just checked for the presence of all necessary fields. So now it only fails if it does not find either of the conditions needed for either a disk or Fermi instrument.